### ISSUE TITLE
Branch 7.0 not viable?

### ISSUE THREAD

**Issue by jonathan-dejong** [mpdf/mpdf#159]
Hi,

Are branch 7.0 not ready for being used yet?
I tried it now and it contains sooooo much php errors. All related to wrongly coded statements like 
`
if(blabla){
$oneliner = 'but there is no end bracket';

`

I sifted through it trying to fix them but 30k + lines are too much. 
Am I missing something or..?

**Comment by jonathan-dejong** [comment]
Just realized I also ran it through the compressor. Perhaps that's the issue.. ?

**Comment by finwe** [comment]
Branch 7.0 is now even with 6.1 (with two minor commits). So it should work the same. So I guess the compressor (if you mean the `compress.php`) script is the issue. It is meant to be removed in the 7.0 version.

**Comment by jonathan-dejong** [comment]
@finwe Thanks for the quick reply.
Yeah I tried the master branch which I'm guessing is pretty much equal to 7.0 and it worked without running it through `compressor.php`.

While I have you I might as well shamefully ask you something else. 
I've been using mpdf to batch create pdfs (~250). Recently they started pushing the memory with 5mb per iteration. I've fixed all leaks in my code so with the master branch version it's now down to 1 - 1.5mb per iteration however it's still a lot more than it used to be (somehow it's increased). The PDF's created are single page and very small (around 50k with just some simple text, table and a jpg logo). 

Are there any way to get this down further? Could there be some memory leaks to fix in mpdf? Checking memory usage the increase in memory happens during/after the `$mpdf->Output('f')` function. It's basically not moving before that after I've fixed my own leaks so it's entirely happening in there.
`
error_log('Memory used: ' . (memory_get_peak_usage(true)/1024/1024) .'mb');
		$mpdf->Output($pdf_path, 'F');
		unset($mpdf);
		gc_collect_cycles();
		error_log('Memory used: ' . (memory_get_peak_usage(true)/1024/1024).'mb' );
`

**Comment by finwe** [comment]
Has the code started pushing the memory after a concrete release? Or do you know what was the cause of the memory usage growth?

I am sure that there are many points in mPDF code that could be optimized, that surely can be done when a global code cleanup is done in version 7.0

**Comment by jonathan-dejong** [comment]
That's the strange part. It's been working flawlessly with v 6.0 in production (and dev) and I've generated the PDFs many times during development without issue (having 128mb memory limit). Then the automatic generation kicked in a few days ago and suddenly it was up to 5mb resulting in memory limit error. 

I've noticed that the hosting provider has updated their admin panel recently (still ancient *sigh*) so perhaps there's been some changes in server software that could be the culprit. However I'm not sure I can do anything about it since the client is not willing to switch (and I'm just a "sub consultant" for another agency). I've checked phpinfo tho and it's still on 128mb memory and from what I can tell standard libs are installed.
